src/testdrive/src/action/consistency.rs

```diff
@@ -152,6 +148,32 @@ async fn check_catalog_state(state: &State) -> Result<(), anyhow::Error> {
     .await
     .context("deserialize catalog")?;
 
+    // Pull out the system parameter defaults from the in-memory catalog, as we
+    // need to load the disk catalog with the same defaults.
+    let dump: CatalogDump = serde_json::from_str(&memory_catalog).context("decoding catalog")?;
+
+    let Some(system_parameter_defaults) = dump.system_parameter_defaults else {
+        // TODO(parkmycar, def-): Ideally this could be an error, but a lot of test suites fail. We
+        // should explicitly disable consistency check in these test suites.
+        tracing::warn!(
+            "Missing system_parameter_defaults in memory catalog state, skipping consistency check"
+        );
```

Comment:
Which tests fail? It's weird that a test is trying to validate the catalog state but is unable to to dump it.

Reply:
Upgrade checks because the old Mz version doesn't have system_parameter_defaults yet. But in a few versions this will work in most upgrade tests.